File: Java/Q32.java
import java.util.Scanner;

public class PrimeNumbersInInterval {

    // Function to check if a number is prime
    static boolean isPrime(int num) {
        if (num <= 1)
            return false;

        for (int i = 2; i <= Math.sqrt(num); i++) {
            if (num % i == 0)
                return false;
        }
        return true;
    }

    // Function to display prime numbers in a given range
    static void displayPrimes(int start, int end) {
        System.out.println("Prime numbers between " + start + " and " + end + " are:");
        for (int i = start; i <= end; i++) {
            if (isPrime(i)) {
                System.out.print(i + " ");
            }
        }
    }

    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);

        // Input range
        System.out.print("Enter the starting number: ");
        int start = scanner.nextInt();

        System.out.print("Enter the ending number: ");
        int end = scanner.nextInt();

        // Function call
        displayPrimes(start, end);

        scanner.close();
    }
}

File: Python/Q32.py
# Q32.py - Program to Display Prime Numbers Between Intervals Using Function
def is_prime(n):
    if n <= 1: return False
    for i in range(2, int(n**0.5)+1):
        if n % i == 0:
            return False
    return True

start = int(input("Enter start: "))
end = int(input("Enter end: "))
print(f"Prime numbers between {start} and {end}:")
for i in range(start, end+1):
    if is_prime(i):
        print(i, end=' ')
print()